packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -277,7 +278,30 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
     .toList();
   }
 
-  Future<void> _validateIconAssetsAfterArchive(StringBuffer messageBuffer) async {
+  ValidationResult? _createValidationResult(String title, List<ValidationMessage> messages) {
+    if (messages.isEmpty) {
+      return null;
+    }
+    final bool anyInvalid = messages.any((ValidationMessage message) => message.type != ValidationMessageType.information);
+    return ValidationResult(
+      anyInvalid ? ValidationType.partial : ValidationType.installed,
```

Comment:
@christopherfujino are you concerned the doctor validator is being used outside of the doctor (ex. this is called "installed")?

Reply:
I think it makes sense to either

(1) Define our own types, since there are multiple other `ValidationType`s that are not applicable at all here; 

(2) Alternatively we can also rename `.installed` to something like `.passed`. 

I am fine either way.

Reply:
@christopherfujino friendly reminder

Reply:
I would prefer option 1. I think re-using these types of trivial enums/classes tends to lead over time to certain enum values only making sense in certain contexts, and it becomes difficult to reason about. There's already prior art of us copy pasting this for the `flutter analyze --suggestions` sub-command: https://github.com/flutter/flutter/blob/main/packages/flutter_tools/lib/src/project_validator_result.dart

Reply:
We copy it everywhere instead of sharing it?  This guidance is coming from UX, if we have a list of details and warnings and errors than make it look like the doctor.

Reply:
Ah @jmagman I just saw your comment. I actually created those private types (e.g. `_ValidationResult`), but i'm fine reverting too

Reply:
> How would that work though? This code is using all the doctor `ValidationResult` code, which needs the type to know how to print it. For example
> 
> https://github.com/flutter/flutter/blob/47b40601d7c5228d19630f14144d864bd6065fc1/packages/flutter_tools/lib/src/doctor_validator.dart#L167-L180

ahh, I missed that. re-using it and renaming `installed` -> `success` SGTM